controllers/spiaccesstoken_controller.go

```diff
@@ -272,6 +276,18 @@ func (r *SPIAccessTokenReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return ctrl.Result{RequeueAfter: r.durationUntilNextReconcile(&at)}, nil
 }
 
+func (r *SPIAccessTokenReconciler) reconcileTokenData(ctx context.Context, at *api.SPIAccessToken) error {
+	if at.Status.TokenMetadata != nil && at.Status.Phase == api.SPIAccessTokenPhaseReady {
```

Comment:
Can we do this somehow without depending on the status? Or have a thorough explanation why doing it this way is ok?

Maybe it is, but I'm fearing we may be making the reconciliation not idempotent by these status checks.

Reply:
technically it does not depend on state. This is just an optimization to not check the data for all tokens, but only for those which we expect to have some data (ready phase + metadata!=nil). I can remove this condition and it will work the same, but we will touch tokenstorage everytime, even in cases we expect no data there.

I'm not sure which option is more correct.

Reply:
If the behavior is the same without this check and it is only an optimization then that's great. But if that's the case, I would prefer an explanation of the fact here, because it is not obvious at a glance (at least for me).

Reply:
added comments